### DISCUSSION TITLE
Carriage return does not work in run-shell-command

### DISCUSSION THREAD

**Discussion by KevinGimbel** [helix-editor/helix#11690] [General]
### Summary

When running a shell command with `run-shell-command` which uses carriage return (`\r`) the lines are printed in one line.

### Reproduction Steps

 

Recording https://asciinema.org/a/Nl9KT1Wk5RH90UGBPIZjGob5H 

I tried this:

1. `hx`
2. `:run-shell-command` + my script which prints in a loop with `\r`

I expected this to happen:
- Carriage return (`\r`) should return the cursor to the beginning of the line and replace the line

Instead, this happened:
- The line was written without carriage return

The same happens with a small bash script

```bash
#!/bin/bash

for i in 1 2 3 4 5; do
  printf "\r Iteration $i"
  sleep 1
done
```

<img width="583" alt="Screenshot 2024-09-13 at 09 47 26" src="https://github.com/user-attachments/assets/0f5657d5-522b-4d5b-844d-dcb4a92f0710">

### Helix log

<details><summary>~/.cache/helix/helix.log</summary>

```
please provide a copy of `~/.cache/helix/helix.log` here if possible, you may need to redact some of the lines
```

</details>

### Platform

macOS

### Terminal Emulator

wezterm 20240203-110809,5046fc22

### Installation Method

brew

### Helix Version

helix 24.7 (079f5442)

**Comment by RoloEdits** [comment]
Helix doesn't have a "shell" that would "interpret" the returns(or anything else), its just displaying the returned output as a string; the raw output is captured and returned after the end of shell execution.

Something basic could probably be done for this specific case, where if `\r` is not followed by a `\n` it can truncate as if that was the new start of the string. Done in a loop this would yield only the last result. I'm not sure what kind of functionality you are looking for, but `sh` is only meant for most simple feedback use cases., not as a shell in itself.

**Reply by KevinGimbel** [reply]
Thanks! I wasn't looking for anything specific, I just assumed it would use a "real shell" (for lack of better wording).
If this is intended behaviour I'm fine with it.
